src/main/scala/mesosphere/marathon/core/task/tracker/impl/InstanceTrackerActor.scala

```diff
@@ -27,14 +28,17 @@ import scala.util.{Failure, Success}
 
 object InstanceTrackerActor {
   def props(
-    metrics: ActorMetrics,
-    taskLoader: InstancesLoader,
-    updateStepProcessor: InstanceTrackerUpdateStepProcessor,
-    stateOpResolver: InstanceUpdateOpResolver,
+    metrics: Metrics,
+    config: InstanceTrackerConfig,
+    steps: Seq[InstanceChangeHandler],
     repository: InstanceView,
     clock: Clock,
-    crashStrategy: CrashStrategy): Props = {
-    Props(new InstanceTrackerActor(metrics, taskLoader, updateStepProcessor, stateOpResolver, repository, clock, crashStrategy))
+    crashStrategy: CrashStrategy)(implicit mat: Materializer): Props = {
+    val instancesLoader = new InstancesLoaderImpl(repository, config)(mat)
+    val updateStepProcessor = new InstanceTrackerUpdateStepProcessorImpl(metrics, steps)
+    val stateOpResolver: InstanceUpdateOpResolver = new InstanceUpdateOpResolver(clock)
```

Comment:
Hm, so technically one cannot inject a different `InstanceTrackerUpdateStepProcessor` or `InstanceUpdateOpResolver` implementation anymore, right? The question is if we want to do that.